File: src/main/java/us/achromaticmetaphor/imcktg/ChooseFilename.java
package us.achromaticmetaphor.imcktg;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

public class ChooseFilename extends Activity {
  public void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_choose_filename);
    findViewById(R.id.confirm).setOnClickListener(view -> {
      Intent intent = new Intent(this, ConfirmContacts.class);
      intent.putExtra("forDefault", true);
      intent.putExtra("toneString", ((TextView) findViewById(R.id.def_ringtone_text_box)).getText().toString());
      intent.putExtra("filename", ((TextView) findViewById(R.id.filename_text_box)).getText().toString());
      startActivity(intent);
      finish();
    });
  }
}

File: src/main/java/us/achromaticmetaphor/imcktg/ChooseFilename.kt
package us.achromaticmetaphor.imcktg

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.TextView

class ChooseFilename : Activity() {
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_choose_filename)
        findViewById<View>(R.id.confirm).setOnClickListener {
            startActivity(Intent(this, ConfirmContacts::class.java).apply {
                putExtra("forDefault", true)
                putExtra("toneString", (findViewById<View>(R.id.def_ringtone_text_box) as TextView).text.toString())
                putExtra("filename", (findViewById<View>(R.id.filename_text_box) as TextView).text.toString())
            })
            finish()
        }
    }
}